python/GangaDirac/Lib/Files/DiracFile.py

```diff
@@ -137,7 +137,7 @@ def locations(self):
 
     def _setLFNnamePattern(self, lfn="", namePattern=""):
 
-        if self.defaultSE != "":
+        if hasattr(self, 'defaultSE') and self.defaultSE != "":
```

Comment:
Given that it's part of the schema why is this change necessary? wouldn't it be guaranteed to have this attribute?